include/domain_bridge/topic_bridge_options.hpp

```diff
@@ -141,6 +142,35 @@ class TopicBridgeOptions
   bool
   wait_for_publisher() const;
 
+  enum class AutoRemove
+  {
+    OnNoPublisher,
+    OnNoSubscription,
+    OnNoPublisherOrSubscription,
+    Disabled,
+  };
+
+  /// Set auto_remove to value.
+  /**
+   * If set to Autoremove::OnNoPublisher the domain bridge will stop bridging when no publisher is
+   * available in the "from domain".
+   * The bridge will automatically be created again when a new publisher is discovered.
+   *
+   * If set to Autoremove::OnNoSubscription the domain bridge will stop bridging when no
+   * subscription is available in the "to domain".
+   * The bridge will automatically be created again when a new subscription is discovered.
+   *
+   * When set to Autoremove::Disabled, the bridge will keep running for ever once created.
+   */
+  DOMAIN_BRIDGE_PUBLIC
+  TopicBridgeOptions &
+  auto_remove(AutoRemove value);
```

Comment:
I wonder if having an enum here instead of a boolean is exposing _too_ much configurability. Having a scenario where the "wait for" options are not symmetric with the auto-removal policy seems odd to me, but maybe there's a use-case.

I suppose either way it doesn't really affect the complexity of the implementation, so we can leave it as-is. I was just curious if you could think of a scenario where we want this level of configuration.

Reply:
> Having a scenario where the "wait for" options are not symmetric with the auto-removal policy seems odd to me

Yes, I agree.

> I suppose either way it doesn't really affect the complexity of the implementation, so we can leave it as-is.

:+1:

> I was just curious if you could think of a scenario where we want this level of configuration.

Not really, an independent option was the first implementation that came to mind.